docs/release.md

```diff
@@ -86,12 +83,14 @@ The last step of the release process creates artifacts for the new version of th
 
 4. Check out the commit created by merging the PR and tag with the new release version by running the `make push-tags TAG=v0.85.0` command. If you set your remote using `https` you need to include `REMOTE=https://github.com/open-telemetry/opentelemetry-collector-releases.git` in each command. Wait for the new tag build to pass successfully.
 
-5. Ensure the "Release Core", "Release Contrib" and "Release k8s" actions pass, this will
+5. Ensure the "Release Core", "Release Contrib", "Release k8s", and "Builder - Release" actions pass, this will
 
     1. push new container images to `https://hub.docker.com/repository/docker/otel/opentelemetry-collector`, `https://hub.docker.com/repository/docker/otel/opentelemetry-collector-contrib` and `https://hub.docker.com/repository/docker/otel/opentelemetry-collector-k8s`
 
     2. create a Github release for the tag and push all the build artifacts to the Github release. See [example](https://github.com/open-telemetry/opentelemetry-collector-releases/actions/workflows/release-core.yaml).
 
+    3. build and release ocb binaries under a separate tagged Github release, e.g. `cmd/builder/v0.85.0`
```

Comment:
I'm open to other opinions, but I think I would like to keep everything under a single tag if possible. Right now we have all builder release notes grouped in with our main tag release, so users would grab the binary from one release but read the changelog in another, and that feels odd to me. I also feel like the advantages of having a single tag vs. separate tags are 50/50, so keeping a single tag would be simpler. 

The advantages I see are:
* Separate tags makes the ocb binary easier to find amongst the currently ~383 assets we publish. It also provides some logical separation between distro releases and other binary releases.
* A single tag means the release itself is easier to find (in my opinion) since the GH releases page feels geared toward a single stream of releases rather than concurrent releases of similar things. Also means the binaries and changelogs are together, and is a simpler setup.

Reply:
Can we solve discoverability with some links pointing users to where the artifacts are located? that way we have both the separation (both logical and in terms of the artifacts) as well as discoverability, at the cost of one more click for some people

Reply:
I don't think the separation is super important since we essentially release everything as a single unit anyway, but that seems like a reasonable compromise.

Reply:
I agree that having everything released under one tag would be ideal; I didn't dig deep enough into how the partial goreleaser releases worked for the distributions to implement it for builder, and plus wanted to get others' opinions on if keeping the separate tags was smart, especially if we are adding other things to this release down the line like telemetrygen, etc.

Reply:
I'm still okay with a single tag. Everything is pretty tightly coupled, and I don't think the separate tags/releases will really be something we can leverage in the near future. Plus, adding ~8 binaries on top of the existing 383 doesn't really make a big difference, especially when a new Collector distro will presumably add 50+ new binaries anyway.

That said, I don't really feel strongly about this if we can link to it from the main release, and we can always change it later. I'll bring this up at tomorrow's SIG in case anyone has strong opinions.